include/devicetree/reserved.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file
+ * @brief reserved-memory Devicetree macro public API header file.
```

Comment:
rename this to memory.h.  `reserved` seems too generic.

Reply:
uhm, how `memory.h` is less generic than `reserved.h`? Anyway, I'll rename it.